File: frontend-admin/src/api/license.ts
import { api } from "./client";

// Giấy phép server (AD-74).
export interface LicenseInfo {
  status: "valid" | "trial" | "expired" | "missing" | "clock_tampered";
  issued_to: string | null;
  expires_at: string | null;
  days_left: number | null;
  warn: boolean;
}

export const licenseApi = {
  get: async (): Promise<LicenseInfo> => (await api.get("/admin/license")).data,
  set: async (key: string): Promise<LicenseInfo> =>
    (await api.post("/admin/license", { key })).data,
};

File: backend/app/schemas/license.py
"""Schemas trang Giấy phép (AD-74)."""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field


class LicenseInfo(BaseModel):
    status: str  # valid | trial | expired | missing | clock_tampered
    issued_to: str | None = None
    expires_at: datetime | None = None
    days_left: int | None = None
    # true khi còn ≤ WARN_DAYS ngày — dashboard hiện cảnh báo đỏ.
    warn: bool = False


class LicenseSetIn(BaseModel):
    key: str = Field(min_length=10, max_length=2000)
